src/main/resources/google_checks.xml

```diff
@@ -109,26 +109,26 @@
         </module>
         <module name="ParameterName">
             <property name="id" value="ParameterNameNonPublic"/>
-            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
+            <property name="format" value="^[a-z]([a-z0-9]+[A-Z]?)*$"/>
             <property name="accessModifiers" value="protected, package, private"/>
             <message key="name.invalidPattern"
              value="Parameter name ''{0}'' must match pattern ''{1}''."/>
         </module>
         <module name="ParameterName">
             <property name="id" value="ParameterNamePublic"/>
-            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
+            <property name="format" value="^[a-z]([a-z0-9][A-Z]?)+$"/>
```

Comment:
@romani 
Google style prohibits one-character parameter names for public methods and allows them form non-public methods. That is why we have 2 different regexpses.